tools/sgcloudspanner/emulator.go

```diff
@@ -60,25 +91,27 @@ func RunEmulator(ctx context.Context) (_ func(), err error) {
 		if err := cmd.Run(); err != nil {
 			sg.Logger(ctx).Printf("failed to remove emulator container: %v", err)
 		}
-		if err := os.Unsetenv("SPANNER_EMULATOR_HOST"); err != nil {
-			sg.Logger(ctx).Printf("failed to unset SPANNER_EMULATOR_HOST: %v", err)
-		}
 	}
-	emulatorHost, err := inspectPortAddress(ctx, containerID, "9010/tcp")
+	spannerEmulator.DatabaseHost, err = inspectPortAddress(ctx, containerID, "9010/tcp")
 	if err != nil {
-		cleanup()
 		return nil, err
 	}
-	if err := os.Setenv("SPANNER_EMULATOR_HOST", emulatorHost); err != nil {
-		cleanup()
+	spannerEmulator.AdminHost, err = inspectPortAddress(ctx, containerID, "9020/tcp")
+	if err != nil {
 		return nil, err
 	}
-	sg.Logger(ctx).Printf("running Cloud Spanner emulator on %s", emulatorHost)
-	if err := awaitReachable(ctx, emulatorHost, 100*time.Millisecond, 10*time.Second); err != nil {
-		cleanup()
+	sg.Logger(ctx).Printf("running Cloud Spanner emulator on %s", spannerEmulator.DatabaseHost)
+	if err := awaitReachable(ctx, spannerEmulator.DatabaseHost, 100*time.Millisecond, 10*time.Second); err != nil {
 		return nil, err
 	}
-	return cleanup, nil
+	return spannerEmulator, nil
+}
+
+func (e *Emulator) Close() {
+	if e == nil || e.onClose == nil {
```

Comment:
it should be possible to write the API so that these types of nil checks are not necessary - when would these happen?

Reply:
Let me remove those, just added them for safety